src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -381,7 +384,23 @@ public SCMRevision run(GitClient client, String remoteName) throws IOException,
                                           // tag does not exist
                                           return null;
                                       }
+                                  } else if (head instanceof GitBranchSCMHead) {
```

Comment:
`|| !(head instanceof GitSCMHeadMixin)`

because if regular git (i.e. you didn;t override this method) then we can only have tags or branches so a head that is "old style" ie a raw `SCMHead` (or a subclass before the mix-in was added to the plugin) can be assumed to be a `GitBranchSCMHead`